diffyscan/utils/encoder.py

```diff
@@ -0,0 +1,109 @@
+from .logger import logger
+
+
+def encode_address(address):
+    number = int(address, 16)
+    formatted_address = f"{number:064x}"
+    return formatted_address
+
+
+def encode_uint256(number):
+    return format(number, "064x")
+
+
+def encode_bytes32(data):
+    return data.replace("0x", "")
+
+
+def encode_bytes(data):
+    bytes_str = data.lstrip("0x")
+    data_length = len(bytes_str) // 2
+    encoded_length = 0
+    if data_length > 0:
+        encoded_length = ((len(bytes_str) - 1) // 64 + 1) * 64
+    bytes_str += "0" * (encoded_length - len(bytes_str))
+    return format(data_length, "064x") + bytes_str
+
+
+def encode_tuple(types, args):
+    args_length = len(types)
+    encoded_offsets = ""
+    encoded_data = ""
+    for arg_index in range(args_length):
+        arg_type = types[arg_index]
+        arg_value = args[arg_index]
+        if arg_type == "address":
+            encoded_offsets += encode_address(arg_value)
+        elif arg_type == "uint256" or arg_type == "bool" or arg_type == "uint8":
+            encoded_offsets += encode_uint256(arg_value)
+        elif arg_type == "bytes32":
+            encoded_offsets += encode_bytes32(arg_value)
+        elif arg_type == "address[]" and not arg_value:
+            encoded_data += "0" * 64
+            offset = format((arg_index + args_length) * 32, "064x")
+            encoded_offsets += offset
+        else:
+            logger.warn(
+                f"Unknown constructor argument type '{arg_type}', use --constructor-calldata instead"
```

Comment:
I can't find any `--constructor-calldata` except this place, how should this work?

Reply:
done. Added field 'constructor_calldata' into config for using prepared calldata, see updates into sepolia and holesky configs